src/models.ts

```diff
@@ -0,0 +1,119 @@
+import { computeMapColors } from "./colormaps/colormapUtils";
+import { ColorInterpolateName } from "./colormaps/d3ColorSchemes";
+
+interface SurfaceMeshSettings {
+  vertices: Float32Array;
+  faces: Uint32Array;
+}
+
+interface VertexMapSettings {
+  intensity: number[];
+  colorLimits: [number, number];
+  colorMapName: ColorInterpolateName;
+}
+
+/**
+ * Represents a surface mesh with vertices and faces.
+ */
+export class SurfaceMesh {
+  private vertices: Float32Array;
+  private faces: Uint32Array;
+
+  /**
+   * Creates a new SurfaceMesh object.
+   * @constructor
+   * @param {Float32Array} vertices - The vertices of the surface model.
+   * @param {Uint32Array} faces - The faces of the surface model.
+   */
+  constructor(vertices: Float32Array, faces: Uint32Array) {
+    this.vertices = vertices;
+    this.faces = faces;
+  }
+
+  /**
+   * Getter for the data of surface mesh properties.
+   * @param propertyName - The name of the property to retrieve.
+   * @returns The Float32Array or Uint32Array corresponding to the given property name.
+   */
+  get(propertyName: "vertices"): Float32Array;
+  get(propertyName: "faces"): Uint32Array;
+  get(propertyName: "vertices" | "faces") {
+    return this[propertyName];
```

Comment:
Why do these have to be properties? Why not either `public` or `readonly`?

Reply:
I chose this because I wanted it to be `readonly`, but `readonly` also prevents private writing. This was the workaround I found towards that end; would you just make them public?

Reply:
Oh yeah seems like it's a missing language feature https://github.com/microsoft/TypeScript/issues/37487

I'm fine with whatever then. Probably I would prefer just `obj.getIntensity()` for auto complete convenience but do what you think is best.